analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -41,6 +41,7 @@ import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
 import org.ossreviewtoolkit.model.config.RepositoryConfiguration
 import org.ossreviewtoolkit.utils.CommandLineTool
 import org.ossreviewtoolkit.utils.Os
+import org.ossreviewtoolkit.utils.log
```

Comment:
commit message: It says "which are
not directly used by the main module". I believe using the term "directly" is unclear. What do you mean by that?

Reply:
Ah, looks like I had a similar question [above](https://github.com/oss-review-toolkit/ort/pull/4119/commits/75e835073286deeed6e3d17c843833428c493aed#r645843909).

Reply:
This is a good question. Unfortunately, my knowledge of go modules is quite limited, and I was mainly following the advice given in #3934. As I understand it from the description in https://golang.org/ref/mod#modules-overview, a module can consist of multiple packages. The dependency graph per default seems to include all the packages of a module. `go mod why` obviously inspects the `import` statements starting from the main module to determine whether specific packages are actually referenced.

I slightly reworded the commit message to align to the terminology used by `go mod why`. The command prints "main module does not need package XXX" for the packages that are now filtered out.